### PR TITLE
[Feature] Support function roundBankers

### LINKED ISSUE #14970: [Feature] Support function `roundBankers`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

ClickHouse supports function `roundBankers` but Doris doesn't support now

If you are interested in it, you could leave a comment for me.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by maochongxin**
# Proposed changes

Issue Number: close #14970

## Problem summary

support roundBankers

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.83 seconds
 load time: 473 seconds
 storage size: 17123356936 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221217143656_clickbench_pr_64679.html

**Review by Gabriel39 — COMMENTED** [review]
Great job! Could you also add some documents for this new function?

**Comment by maochongxin** [comment]
> Great job! Could you also add some documents for this new function?

Ok, let me do this

File: docs/en/docs/sql-manual/sql-functions/math-functions/round_bankers.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+{
+    "title": "roundBankers",
```
Comment (Gabriel39) [inline docs/en/docs/sql-manual/sql-functions/math-functions/round_bankers.md:3]: Also add this to `sidebar.json`

**Review by Gabriel39 — DISMISSED** [review]
LGTM. Thanks for your contribution!

**Comment by Gabriel39** [comment]
There are some format problems and you need to fix them before merging

**Comment by maochongxin** [comment]
> There are some format problems and you need to fix them before merging

fixed

File: docs/en/docs/sql-manual/sql-functions/math-functions/round_bankers.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+specific language governing permissions and limitations
+under the License.
+-->
+
+## roundBankers
+
+### description
+#### Syntax
+
+`roundBankers(x), roundBankers(x, d)`
```
Comment (morningman) [inline docs/en/docs/sql-manual/sql-functions/math-functions/round_bankers.md:30]: the function name should be `round_bankers`, not `roundBankers`

File: docs/zh-CN/docs/sql-manual/sql-functions/math-functions/round_bankers.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+---
+{
+    "title": "roundBankers",
```
Comment (Gabriel39) [inline docs/zh-CN/docs/sql-manual/sql-functions/math-functions/round_bankers.md:3]: roun_bankers

**Comment by Gabriel39** [comment]
Do you still want to update this PR? If not, maybe you should make this PR ready for review

**Comment by maochongxin** [comment]
> Do you still want to update this PR? If not, maybe you should make this PR ready for review

I am currently using mysql-client to connect to fe locally, but I cannot use round_bankers. I am looking into this problem, and can you reproduce it?

**Comment by Gabriel39** [comment]
> I am currently using mysql-client to connect to fe locally, but I cannot use round_bankers. I am looking into this problem, and can you reproduce it?

I think you forget to register this function in FE's FunctionSet. You can add it in gensrc/script/doris_builtin_functions.py just like function round.

Btw, you'd better to add a SQL-like regression test case in folder regression/suites

**Comment by maochongxin** [comment]
> > I am currently using mysql-client to connect to fe locally, but I cannot use round_bankers. I am looking into this problem, and can you reproduce it?
> 
> I think you forget to register this function in FE's FunctionSet. You can add it in gensrc/script/doris_builtin_functions.py just like function round.
> 
> Btw, you'd better to add a SQL-like regression test case in folder regression/suites

That's right!! I did trace the code and found that `getBuiltinFunction` returned `nil`, so it was registered through here, thank you very much! And I will complete the test cases as soon as possible

File: docs/en/docs/sql-manual/sql-functions/math-functions/round_bankers.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+mysql> select round_bankers(1667.2725, -2);
++------------------------------+
+| round_bankers(1667.2725, -2) |
++------------------------------+
+|                         1700 |
++------------------------------+
+```
+
+### keywords
+	ROUNDBANKERS
```
Comment (Gabriel39) [inline docs/en/docs/sql-manual/sql-functions/math-functions/round_bankers.md:80]: round_bankers

File: gensrc/script/doris_builtins_functions.py [hunk 5]
```diff
@@ -1831,24 +1831,32 @@
...
             '_ZN5doris13MathFunctions5roundEPN9doris_udf15FunctionContextERKNS1_9DoubleValE', '', '', 'vec', ''],
+    [['round_bankers'], 'DOUBLE', ['DOUBLE'],
+            '__ZN5doris13MathFunctions13round_bankersEPN9doris_udf15FunctionContextERKNS1_9DoubleValE', '', '', 'vec', ''],
     [['ceil', 'ceiling', 'dceil'], 'DECIMAL32', ['DECIMAL32'],
             '_ZN5doris13MathFunctions4ceilEPN9doris_udf15FunctionContextERKNS1_9DoubleValE', '', '', 'vec', ''],
     [['floor', 'dfloor'], 'DECIMAL32', ['DECIMAL32'],
             '_ZN5doris13MathFunctions5floorEPN9doris_udf15FunctionContextERKNS1_9DoubleValE', '', '', 'vec', ''],
     [['round', 'dround'], 'DECIMAL32', ['DECIMAL32'],
             '_ZN5doris13MathFunctions5roundEPN9doris_udf15FunctionContextERKNS1_9DoubleValE', '', '', 'vec', ''],
+    [['round_bankers'], 'DECIMAL32', ['DECIMAL32'],
```
Comment (Gabriel39) [inline gensrc/script/doris_builtins_functions.py:1842]: DECIMAL64 and DECIMAL128 are needed

**Comment by Gabriel39** [comment]
You also need to update the result file in regression/data after adding a test SQL. Pls refer to https://doris.incubator.apache.org/zh-CN/community/developer-guide/regression-testing for more details.

**Comment by maochongxin** [comment]
I've finished, help code review @Gabriel39

**Comment by Gabriel39** [comment]
> I've finished, help code review @Gabriel39

Some bugs are fixed in master branch recently but still exist in your dev branch. Could you rebase the master branch and push again?

**Comment by maochongxin** [comment]
> > I've finished, help code review @Gabriel39
> 
> Some bugs are fixed in master branch recently but still exist in your dev branch. Could you rebase the master branch and push again?

emmm, just used sync fork at my source repository

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Comment by Gabriel39** [comment]
Merged. Thanks for your contribution!

**Pull request merged**